### DISCUSSION TITLE
[Multiplayer, Campaign] Multiple characters per player

### DISCUSSION THREAD

**Discussion by Dave247** [FakeFishGames/Barotrauma#9766] [Ideas and suggestions]
Have it so players can create and campaigns can store a limited number (eg. 3-5) separate characters in a campaign that they can switch between rounds, so that they can switch up different roles to keep things interesting, or just to make sure each role is filled if one person has to leave early but will be back again to play tomorrow.

**Comment by Jamessy22** [comment]
I would actually suggest to make it more akin to "single-player" management: a set of "characters" that one can buy at a station, then assign players to control them. Ideally, at any given time each player could switch between a number of "assigned" characters - and AI would take over any characters that aren't currently being controlled.

Bottomline is, with current "fresh character per player" approach, resources get lost/stuck in savefile when a player leaves or disconnects - which is particularly problematic following a server crash, when some people don't rejoin. The loss of skills is very tedious as well, joining a campaign in hydrothermal wastes is a bit problematic when you spawn as level one character.

This could be implemented as a separate "mode" to be selected when starting a new campaign, or switched in lobby if technically feasible. Other than that I reckon being able to "assign" only one character per person would be a decent starting point, then expanding it if possible/desired? Or I don't know, does this idea make any sense at all?

**Reply by Nonamow** [reply]
There is some mods https://steamcommunity.com/sharedfiles/filedetails/?id=2775613786

But yes. At least, I want to be able to control my bots like in solo.

> Ideally, at any given time each player could switch between a number of "assigned" characters - and AI would take over any characters that aren't currently being controlled.

That would be great.

**Comment by erunehtar** [comment]
Hi, I implemented the feature directly in the game source code rather than a mod. Changes can be found in the branch here: https://github.com/erunehtar/Barotrauma/tree/allow-controlling-bots-in-multiplayer if you want to try it.

**Reply by erunehtar** [reply]
pull request: https://github.com/FakeFishGames/Barotrauma/pull/17843
